### ISSUE TITLE
helm repo index should support a switch to choose the local output filename

### ISSUE THREAD

**Issue by wknapik** [helm/helm#2804]
At the moment, the filename is hardcoded to index.yaml.

**Comment by bacongobbler** [comment]
that would be because all helm repo-related commands reference index.yaml as the page to fetch the repo metadata.

Is there a particular use case this is trying to solve?

**Comment by wknapik** [comment]
Well, here's one possible case - if you want to generate two differently named index files, based on the contents of the same directory, in parallel (e.g. make -j), then it creates a race condition, because you actually generate the same file and then move. You can work around it in a 1000 ways, but that's backwards.

Honestly though, does this require a specific use case ? Hardcoding paths on systems you don't control is Wrong. It makes life more difficult for everyone for no good reason.

**Comment by wknapik** [comment]
If this is a case of WONTFIX, feel free to close it, otherwise
/remove-lifecycle stale

**Comment by bacongobbler** [comment]
closing as inactive.
